src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -217,10 +217,18 @@ public bool ReportErrors(IEnumerable<Diagnostic> diagnostics, TextWriter console
                     // Not reported from the command-line compiler.
                     continue;
                 }
-
-                consoleOutput.WriteLine(DiagnosticFormatter.Format(diag, this.Culture));
+                
                 ErrorLogger.LogDiagnostic(diag, this.Culture, errorLogger);
 
+                // We want to report diagnostics with source suppression in the error log file.
+                // However, these diagnostics should not be reported on the console output.
+                if (diag.HasSourceSuppression)
```

Comment:
Well, we don't even know if people want to look at suppressed diagnostics. We want to make the information available publically but not pollute an important property such as Severity with this data. Also note that once we make it a compilation option of whether or not to report such diagnostics, the host explicitly needs to set this option (false by default) and so indeed knows how to distinguish between source suppressed diagnostics. Design team decided a separate bool property is the best approach.